### DISCUSSION TITLE
Confuse about fast_partition and global_partition, what's diff between them?

### DISCUSSION THREAD

**Discussion by gavin838** [pytorch/TensorRT#3325] [Q&A]
I read the code and i think both functions are the same!
First, both them get supported op from same path, although using different logic. I think they can use unify function, now they are redundant code.
Second, both interfaces find a Group(contain 1 node) firstly, and then accept more nodes topologically which supported by the Backend continually.
Third, fine tune generated subgraph by some config.

**Comment by gavin838** [comment]
Global partition want to find better Group by more global info, but it seem no global info. For example, using fast pratition can split origin Module(A,B,C,D) into (AB)(CD), and global partition split into (A)(BCD) or (AC)(BD), but no evaluation say which partition carry more  performance earnings, like B's output have less data transform or BCD can fuse better.

**Comment by narendasan** [comment]
Most of the time they are the same, but the adjacency partitioner is hueristic based so its faster but not guaranteed to be optimal as in terms of creating the minimal number of subgraphs.
